pkg/ruler/manager.go

```diff
@@ -53,6 +53,8 @@ type DefaultMultiTenantManager struct {
 	configUpdatesTotal            *prometheus.CounterVec
 	registry                      prometheus.Registerer
 	logger                        log.Logger
+
+	rulerIsRunning bool
```

Comment:
It's not clear to me if this is read and written from multiple goroutines. To be safe, this should be an `atomic.Bool` (from the uber sync package).

Reply:
The bool would be changed once and protected by the userManagerMtx, But `atomic.Bool` seems like a good practice for miss using this bool in the future. Thanks for review! ❤️